### ISSUE TITLE
Settings not populated with defaults in nested keys

### ISSUE THREAD

**Issue by clason** [latex-lsp/texlab#707]
It seems that Texlab fails to use the default values of nested keys (e.g., `texlab.build.executable`) if the containing key (e.g., `texlab.build`) is omitted.

Consider the following example (with latest Neovim master)

1. Open any `tex` document.
2. Start texlab (assuming it is available in your PATH) with
```
:lua vim.lsp.start { cmd = 'texlab', settings = { texlab = {  }  } }
```
3. Build the document with
```
:lua vim.lsp.buf_request(0, 'textDocument/build', { textDocument = { uri = vim.uri_from_bufnr(0) }}, function() end)
```
Observe that nothing happens (the texlab log says `ERROR - Build failed: No such file or directory (os error 2)`).

Now repeat the above but start texlab via
```
:lua vim.lsp.start { cmd = 'texlab', settings = { texlab = { build = { } } } }
```
Observe that now the document is built.

**Comment by pfoerster** [comment]
Thanks for the detailed report. This should be fixed with c186e919024c3b0205c04ad1d0177ae0ba6da28b. The issue was a faulty `Default` implementation (which was derived by the compiler). `texlab` used the the empty string as the executable name in this case.

**Comment by clason** [comment]
Ideed, it's fixed now! Thank you!
